docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/set-metadata.md

```diff
@@ -0,0 +1,121 @@
+| ------------------ | ---------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `updates`          | `[]object` | Y        | List of metadata updates to apply to various resources.                                                                                                                                                      |
+| `updates[].kind`   | `string`   | Y        | Kind of resource to update metadata for. Must be either "`Stage`" or "`Freight`".                                                                                                                            |
+| `updates[].name`   | `string`   | Y        | Name of the resource to update metadata for.                                                                                                                                                                 |
+| `updates[].values` | `object`   | Y        | Key-value pairs to set as metadata on the resource. Must contain at least one key-value pair. This field supports various types of values including strings, numbers, booleans, arrays, and nested objects. |
+
+## Examples
+
+### Common Usage
+
+In this example, metadata is added to both a `Stage` and a `Freight` resource. This
+pattern is commonly used to track deployment information or add context about the
+promotion process.
+
+```yaml
+steps:
+  - uses: set-metadata
+    config:
+      updates:
+        - kind: Stage
+          name: production
+          values:
+            deployedBy: "user@example.com"
+            version: "1.0.0"
+        - kind: Freight
+          name: my-app-freight
+          values:
+            deployed: true
+            deploymentStatus: "success"
+```
+
+### Complex Metadata
+
+This example demonstrates setting more complex metadata structures, including nested
+objects and arrays:
+
+```yaml
+steps:
+  - uses: set-metadata
+    config:
+      updates:
+        - kind: Stage
+          name: staging
+          values:
+            deployment:
+              timestamp: "xyz"
+              components:
+                - name: "frontend"
+                  version: "2.1.0"
+                  status: "healthy"
+                - name: "backend"
+                  version: "1.5.0"
+                  status: "healthy"
+            environment:
+              region: "us-west-2"
+              cluster: "staging-01"
+```
+
+### Using Metadata in Other Contexts
+
+Once metadata is set (as shown in the examples above), it can be retrieved using
+[`freightMetadata()`](../40-expressions.md#freightmetadatafreightname) and
+[`stageMetadata()`](../40-expressions.md#stagemetadatastagename) expressions in other
+steps or verification processes. To access the metadata in different contexts:
```

Comment:
```suggestion
Once metadata is set (as shown in the examples above), it can be retrieved using
[`freightMetadata()`](../40-expressions.md#freightmetadatafreightname) and
[`stageMetadata()`](../40-expressions.md#stagemetadatastagename) functions
available within expressions used in other steps or in verification processes.
```